actionpack/lib/action_controller/metal/rate_limiting.rb

```diff
@@ -52,20 +86,34 @@ module ClassMethods
       #       rate_limit to: 3, within: 2.seconds, name: "short-term"
       #       rate_limit to: 10, within: 5.minutes, name: "long-term"
       #     end
-      def rate_limit(to:, within:, by: -> { request.remote_ip }, with: -> { head :too_many_requests }, store: cache_store, name: nil, **options)
+      def rate_limit(to:, within:, by: :rate_limit_by, with: :rate_limit_with, store: cache_store, name: nil, **options)
         before_action -> { rate_limiting(to: to, within: within, by: by, with: with, store: store, name: name) }, **options
       end
     end
 
     private
       def rate_limiting(to:, within:, by:, with:, store:, name:)
-        cache_key = ["rate-limit", controller_path, name, instance_exec(&by)].compact.join(":")
+        callable = ->(c) { c.is_a?(Symbol) ? send(c) : instance_exec(&c) }
+
+        cache_key = ["rate-limit", controller_path, name, callable.call(by)].compact.join(":")
```

Comment:
We can get the same Symbol support if we pass in `self` as the context argument:

```suggestion
        cache_key = ["rate-limit", controller_path, name, instance_exec(self, &by)].compact.join(":")
```

```ruby
str = "yo" # => "yo"
str.instance_exec(str) { _1.upcase } # => "YO"
str.instance_exec(str) { upcase } # => "YO"
str.instance_exec(str, &:upcase) # => "YO"
blk = -> { _1.upcase }
str.instance_exec(str, &blk) # => "YO"
```

Reply:
I like the appeal of this technique! Unfortunately, I wasn't able to get this to work the way I expected with the diff you've provided. Am I missing a step?

Reply:
Huh, I just tried it out locally and yeah, it's not quite working. So I don't think we actually can use this.